org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/ProjectCommand.java

```diff
@@ -744,8 +747,16 @@ public static void updateProjectSettings(String projectUri, Map<String, Object>
 					}
 					break;
 				default:
+					if (validOptionKeys.contains(entry.getKey())) {
+						newOptions.put(entry.getKey(), entry.getValue().toString());
+					}
 					break;
 			}
 		}
+		if (!newOptions.isEmpty()) {
+			Map<String, String> javaProjectOptions = javaProject.getOptions(false);
+			javaProjectOptions.putAll(newOptions);
```

Comment:
@snjeza I remember in a previous PR, you mentioned `JavaCore.setOptions(..)` would trigger a rebuild ? Is this an issue here as well ? Is it worth checking if any of the new options are different than the existing values ?

Reply:
Yes, it is.

Reply:
Thank you. I've updated the PR.